From 47a80fdcb642165483f63e51c5e22563cbaa1431 Mon Sep 17 00:00:00 2001
From: Tay Ray Chuan <rctay89@gmail.com>
Date: Thu, 8 Apr 2010 01:40:24 +0800
Subject: [PATCH 16/22] fix examples

 - add LF for responses
 - add pkt-line lengths for smart
 - add flush-pkt (0000) for smart
 - drop SP that followed NUL before capability list
---
 Documentation/technical/http-protocol.txt | 25 +++++++++++++------------
 1 file changed, 13 insertions(+), 12 deletions(-)

diff --git a/Documentation/technical/http-protocol.txt b/Documentation/technical/http-protocol.txt
index 15a08e2..af54cf1 100644
--- a/Documentation/technical/http-protocol.txt
+++ b/Documentation/technical/http-protocol.txt
@@ -203,21 +203,22 @@ The request MUST NOT contain additional query parameters.
    dumb server reply:
    S: 200 OK
    S:
-   S: 95dcfa3633004da0049d3d0fa03f80589cbcaf31	refs/heads/maint
-   S: d049f6c27a2244e12041955e262a404c7faba355	refs/heads/master
-   S: 2cb58b79488a98d2721cea644875a8dd0026b115	refs/tags/v1.0
-   S: a3c2e2402b99163d1d59756e5f207ae21cccba4c	refs/tags/v1.0^{}
+   S: 95dcfa3633004da0049d3d0fa03f80589cbcaf31	refs/heads/maint\n
+   S: d049f6c27a2244e12041955e262a404c7faba355	refs/heads/master\n
+   S: 2cb58b79488a98d2721cea644875a8dd0026b115	refs/tags/v1.0\n
+   S: a3c2e2402b99163d1d59756e5f207ae21cccba4c	refs/tags/v1.0^{}\n
 
    smart server reply:
    S: 200 OK
    S: Content-Type: application/x-git-upload-pack-advertisement
    S: Cache-Control: no-cache
    S:
-   S: ....# service=git-upload-pack
-   S: ....95dcfa3633004da0049d3d0fa03f80589cbcaf31 refs/heads/maint\0 multi_ack
-   S: ....d049f6c27a2244e12041955e262a404c7faba355 refs/heads/master
-   S: ....2cb58b79488a98d2721cea644875a8dd0026b115 refs/tags/v1.0
-   S: ....a3c2e2402b99163d1d59756e5f207ae21cccba4c refs/tags/v1.0^{}
+   S: 001e# service=git-upload-pack\n
+   S: 004895dcfa3633004da0049d3d0fa03f80589cbcaf31 refs/heads/maint\0multi_ack\n
+   S: 0042d049f6c27a2244e12041955e262a404c7faba355 refs/heads/master\n
+   S: 003c2cb58b79488a98d2721cea644875a8dd0026b115 refs/tags/v1.0\n
+   S: 003fa3c2e2402b99163d1d59756e5f207ae21cccba4c refs/tags/v1.0^{}\n
+   S: 0000
 
 Dumb Server Response
 ^^^^^^^^^^^^^^^^^^^^
@@ -297,8 +298,8 @@ Clients MUST first perform ref discovery with
    C: POST $GIT_URL/git-upload-pack HTTP/1.0
    C: Content-Type: application/x-git-upload-pack-request
    C:
-   C: ....want 0a53e9ddeaddad63ad106860237bbf53411d11a7
-   C: ....have 441b40d833fdfa93eb2908e52742248faf0ee993
+   C: 0032want 0a53e9ddeaddad63ad106860237bbf53411d11a7\n
+   C: 0032have 441b40d833fdfa93eb2908e52742248faf0ee993\n
    C: 0000
 
    S: 200 OK
@@ -480,7 +481,7 @@ Clients MUST first perform ref discovery with
    C: POST $GIT_URL/git-receive-pack HTTP/1.0
    C: Content-Type: application/x-git-receive-pack-request
    C:
-   C: ....0a53e9ddeaddad63ad106860237bbf53411d11a7 441b40d833fdfa93eb2908e52742248faf0ee993 refs/heads/maint\0 report-status
+   C: ....0a53e9ddeaddad63ad106860237bbf53411d11a7 441b40d833fdfa93eb2908e52742248faf0ee993 refs/heads/maint\0report-status
    C: 0000
    C: PACK....
 
-- 
1.8.4.rc4.527.g303b16c

